speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/FileDataStorageTests.cpp

```diff
@@ -35,31 +81,327 @@ inline std::string GetTimeAsString(time_t p_date)
+/// @param p_doSteer Whether a Steer decision  should be written
+/// @param p_doBrake Whether a Brake decision  should be written
+/// @param p_doAccel Whether a Accel decision  should be written
+/// @param p_doGear Whether a Gear decision  should be written
+/// @param p_doLights Whether a Lights decision  should be written
+void WriteExpectedDecisions(DecisionTuple& p_decisions, std::ostream& p_expected, bool p_doSteer, bool p_doBrake, bool p_doAccel, bool p_doGear, bool p_doLights)
+{
+    p_expected << "Decisions\n";
+    if (p_doSteer)
+    {
+        p_expected << "SteerDecision\n"
+                   << std::to_string(p_decisions.GetSteer()) << "\n";
+    }
+    if (p_doBrake)
+    {
+        p_expected << "BrakeDecision\n"
+                   << std::to_string(p_decisions.GetBrake()) << "\n";
+    }
+    if (p_doAccel)
+    {
+        p_expected << "AccelDecision\n"
+                   << std::to_string(p_decisions.GetAccel()) << "\n";
+    }
+    if (p_doGear)
+    {
+        p_expected << "GearDecision\n"
+                   << std::to_string(p_decisions.GetGear()) << "\n";
+    }
+    if (p_doLights)
+    {
+        p_expected << "LightsDecision\n"
+                   << std::to_string(p_decisions.GetLights()) << "\n";
+    }
+    p_expected << "NONE\n";
+}
+
 // Values written at the top of a file initialised with the dummy parameters above
 #define DUMMY_INITIALISATION_FILE_ENTRIES ("Player1\n" + GetTimeAsString(timeSimStart) + "\nnotABlackBox.exe\n" + GetTimeAsString(timeBlackBox) + "\nReally just a string\nenvironmentAlsoFake.xml\n0\nName of a dummy environment\n" + std::to_string(INTERVENTION_TYPE_ONLY_SIGNALS) + "\n")
 
-// Run a single test on the data storage system, containing no data except for the driver's name and the zero timestamp.
-TEST(FileDataStorageTests, NoStorageTimestampZero)
+/// @brief Run the FileDataStorage saving only minimal data and ensure the file is correct and located in the correct directory
+/// @param p_numberOfTicks How many ticks to save (default 1)
+void TestNoStorageWithTimestamps(unsigned int p_numberOfTicks = 1)
 {
     // Initialise class, read+write no values
     DataToStore params = {false, false, false, false, false};
 
     GET_DUMMY_TIMES;
-    // Write a file with dummy initialization data, save timestamp 0, and shut down
+    std::stringstream expected;
+
+    // Write a file with dummy initialization data, save timestamp 0 as many times as needed, and shut down
     FileDataStorage fileDataStorage;
-    fileDataStorage.Initialize(params, DUMMY_INITIALISATION_PARAMETERS);
-    fileDataStorage.Save(nullptr, nullptr, 0);
+    std::experimental::filesystem::path actualPath = fileDataStorage.Initialize(params, DUMMY_INITIALISATION_PARAMETERS);
+    expected << DUMMY_INITIALISATION_FILE_ENTRIES;
+
+    for (int i = 0; i < p_numberOfTicks; i++)
+    {
+        fileDataStorage.Save(nullptr, nullptr, 0);
+        expected << "0\n";
```

Comment:
Change all '\n' characters with std::endl as enters are different on windows and linux